src/MSBuild.UnitTests/MSBuildServer_Tests.cs

```diff
@@ -185,13 +193,23 @@ public void BuildsWhileBuildIsRunningOnServer()
             pidOfServerProcess = ParseNumber(output, "Server ID is ");
             _env.WithTransientProcess(pidOfServerProcess);
 
+            using var watcher = new System.IO.FileSystemWatcher(dir!);
+            ManualResetEvent mre = new ManualResetEvent(false);
+            watcher.Created += (o, e) =>
+            {
+                _output.WriteLine($"The marker file {fileName} was created. The build task has been started.");
+                mre.Set();
+            };
+            watcher.Filter = fileName;
+            watcher.EnableRaisingEvents = true;
             t = Task.Run(() =>
             {
                 RunnerUtilities.ExecMSBuild(BuildEnvironmentHelper.Instance.CurrentMSBuildExePath, sleepProject.Path, out _, false, _output);
             });
 
             // The server will soon be in use; make sure we don't try to use it before that happens.
-            Thread.Sleep(1000);
+            mre.WaitOne();
+            _output.WriteLine("It's OK to go ahead.");
```

Comment:
I'd actually move this log line to _before_ the indefinite wait, so if the wait hangs for a very long time we can see that clearly in the log.

Reply:
Instead, I'd like to add one more log before waiting.